### ISSUE TITLE
could you give an example with nginx?

### ISSUE THREAD

**Issue by airdogvan** [nosduco/nforwardauth#4]

**Comment by fancydos** [comment]
I think I have this figured out. I had change the default response code to UNAUTHORIZED:

```
diff --git a/src/main.rs b/src/main.rs
index 21c53e9..0aa4dbd 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -199,7 +199,7 @@ async fn api_forward_auth(req: Request<IncomingBody>) -> Result<Response<BoxBody
     }

     Ok(Response::builder()
-        .status(StatusCode::TEMPORARY_REDIRECT)
+        .status(StatusCode::UNAUTHORIZED)
         .header(LOCATION, location.to_string())
         .body(full(UNAUTHORIZED))
         .unwrap())
```

NGINX's `auth_request` directive considers a response code of 200 success, codes 40x as auth request/failure, and anything else, e.g. a 307 temporary redirect, an error.

From there, the config is pretty straight forward. You will need a server declaration for your AUTH_HOST:

```
server {
    listen       192.0.2.1:443 default_server ssl;
    server_name  auth.example.com;
    ...
    location / {
        proxy_set_header  Host "auth.example.com";
        proxy_set_header  X-Forwarded-Proto https;
        proxy_pass_header Content-Type;
        proxy_pass        http://auth.example.com:3000/;
    }
}
```

And then in your other server declarations you'll need:

```
    location ^~ /auth {
        rewrite              /auth/(.*)   /$1       break;
        proxy_set_header     Host         "auth.example.com";
        proxy_pass_header    Content-Type;
        proxy_pass           http://auth.example.com:3000/;

    }

    error_page 401 = @error401;
    location @error401 {
        return 302 https://auth.example.com/login?r=https://$http_host$request_uri;
    }

    # add the auth_request to anything you wish to be protected
    location / {
        auth_request         /auth;
    }
```

Just make sure you change the IP address, port, and example hostname to match your setup.

**Comment by bt90** [comment]
@nosduco would it be possible to provide a separate path with the 401 logic?

* `/` -> HTTP 302
* `/auth` -> HTTP 401

**Comment by nosduco** [comment]
@bt90 Very possible, but am thinking the best path might be an environment variable (etc, something like `NGINX=true`) and then change the response status code based on that. Thoughts?

**Comment by bt90** [comment]
> If you wanted to support both behaviors in a single instance, you'd need a different route.

Two or more reverse proxies on the same host sounds a bit uncommon?

> You could make `api_forward_auth()` accept another parameter for the HTTP Status Code to be returned, then change it based on the route path.

I kept thinking about the environment variable approach but ultimately this is the better solution IMHO.

Currently it's possible to look at a caddy/nginx/whatever config and tell someone what's most likely wrong with it. If we change response codes based on an environment variable, this gets more complicated as we need to check both configs.

**Comment by fancydos** [comment]
> Two or more reverse proxies on the same host sounds a bit uncommon?

Agree, but doesn't necessarily have to be on the same host.

**Comment by LudovicLachance** [comment]
I am rewriting the app in java with jakarta and the only thing blocking me is the nginx configuration. Is there a nginx example for Authentication Based on Subrequest Result?

https://nginx.org/en/docs/http/ngx_http_auth_request_module.html
From what I have read:
    If the subrequest returns a 2xx response code, the access is allowed. If it returns 401 or 403, the access is denied with the corresponding error code. Any other response code returned by the subrequest is considered an error.

I am a bit lost, I believe reading the traefik config you use would be nice!!

**Comment by nosduco** [comment]
Hey all, I went ahead and implemented a `/nginx` forward auth route and added an nginx example in the `examples` folder. This is released under the 1.4.0 version of nforwardauth. Thank you all for your recommendations and help! <3 Let me know if anyone runs into any further issues or has trouble getting it up running.
